src/muxer/muxer_libav.c

```diff
@@ -172,6 +174,39 @@ lav_muxer_add_stream(lav_muxer_t *lm,
     st->sample_aspect_ratio.num = c->sample_aspect_ratio.num;
     st->sample_aspect_ratio.den = c->sample_aspect_ratio.den;
 
+    if (ssc->es_type == SCT_H264) {
+      if (av_bsf_alloc(lm->bsf_h264_filter, &lm->ctx)) {
+        tvherror(LS_LIBAV,  "Failed to alloc AVBitStreamFilter for h264_mp4toannexb");
+        return -1;
+      }
+    }
+    else if (ssc->es_type == SCT_HEVC) {
+      if (av_bsf_alloc(lm->bsf_hevc_filter, &lm->ctx)) {
+        tvherror(LS_LIBAV,  "Failed to alloc AVBitStreamFilter for hevc_mp4toannexb");
+        return -1;
+      }
+    }
+    else if (ssc->es_type == SCT_VP9) {
+      if (av_bsf_alloc(lm->bsf_vp9_filter, &lm->ctx)) {
+        tvherror(LS_LIBAV,  "Failed to alloc AVBitStreamFilter for vp9_superframe ");
+        return -1;
+      }
+    }
+    else {
+      if (av_bsf_alloc(lm->bsf_h264_filter, &lm->ctx)) {
+        tvherror(LS_LIBAV,  "Failed to alloc AVBitStreamFilter for h264_mp4toannexb");
+        return -1;
+      }
+    }
+    if(avcodec_parameters_copy(lm->ctx->par_in, st->codecpar)) {
+      tvherror(LS_LIBAV,  "Failed to copy paramters to AVBSFContext");
```

Comment:
Here you fail to release the allocation from above, is that intended?

Reply:
actually is worst than that: in case of any error with premature return -1; I don't release .. let me fix this problem :(.

Reply:
Maybe you can use a `goto` to jump to the end  of this function where all the freeing is happening. The return code is then stored in a rc variable and in the end there is a simple "return rc;"

Reply:
done